server/server.go

```diff
@@ -126,26 +121,33 @@ func (cs *cribbageServer) ginPostCreatePlayer(c *gin.Context) {
 }
 
 func (cs *cribbageServer) ginPostCreateInteraction(c *gin.Context) {
-	pID := getPlayerID(c, `playerId`)
+	var cir network.CreateInteractionRequest
+	err := c.ShouldBindJSON(&cir)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+		return
+	}
+	pID := cir.PlayerID
 	if pID == model.InvalidPlayerID {
 		c.String(http.StatusBadRequest, `Needs playerId`)
 		return
 	}
 
 	var mode interaction.Mode
-	switch c.Param(`mode`) {
+	switch cir.Mode {
 	case `localhost`:
 		mode = interaction.Localhost
```

Comment:
Perhaps instead of having a `Mode` field, we just check which field is set and then infer the mode based on that. Does that make sense?

Reply:
Check which field is set on what?